window_manager/worker.js

```diff
@@ -29,49 +33,48 @@ async function updateWindowsFromArray(windows) {
   const matchersPromise = Matcher.loadAll();
 
   const displays = await displaysPromise;
-  const actions = new Map((await actionsPromise)
-                          .map(a => [a.id, a.createUpdate(displays)]));
-  const matchers = await matchersPromise;
 
+  // create a map of action name -> windowUpdate object for only actions valid for the current set of displays:
+  const actions = new Map(
+    (await actionsPromise)
+      .map(a => [a.id, a.createUpdate(displays)])
+      // createUpdate returns null when no matching display.
+      .filter((pair) => pair[1] != null ));
+  console.log('Got valid actions for current displays: ',[...actions.keys()]);
+
+  let matchers = await matchersPromise;
+
+  // Filter out invalid/unknown actions from each matcher
```

Comment:
yes - in theory invalid actions will not be allowed by the validator in the options page, so the error will be raised to the user when the options are saved, rather than an error that it only visible if the user debugs the extension.